htdocs/commande/class/commande.class.php

```diff
@@ -3512,18 +3512,39 @@ public function load_board($user)
 		}
 		$sql .= $clause." c.entity IN (".getEntity('commande').")";
 		//$sql.= " AND c.fk_statut IN (1,2,3) AND c.facture = 0";
-		$sql .= " AND ((c.fk_statut IN (".self::STATUS_VALIDATED.",".self::STATUS_SHIPMENTONPROCESS.")) OR (c.fk_statut = ".self::STATUS_CLOSED." AND c.facture = 0))"; // If status is 2 and facture=1, it must be selected
+		if ($mode == 'toship') {
+			// An order to ship is an open order (validated or in progress)
+			$sql .= " AND c.fk_statut IN (" . self::STATUS_VALIDATED . "," . self::STATUS_SHIPMENTONPROCESS . ")";
+		}
+		if ($mode == 'tobill') {
```

Comment:
The status "to bill" (the order was not billed) is no more saved into the status of the order. Because some company need to have the order "to bill" whatever is the status of shipment (it is a different process) so we must test on 
"facture" = 0 or not to know if it is billed or not.

However some want to the information "To bill" when order is shipped. And some want the information "to bill" as soon as order has been created. Due to 2 different use case, the information "to bill" was not visible on the workboard (because in half case, it is wrong because it depends on the use case).


So i suggest to add an option into module "Order" to ask when the order must be billed:
- As soon as validated   => test "to bill" will be done on  c.facture = 0";
- When an order as been shipped   => test "to bill" will be done the way you suggest  c.fk_statut IN (" . self::STATUS_CLOSED . ") AND c.facture = 0";

Reply:
@atm-maxime I agree with eldy. Would you be willing to rework the pull request so the user get a new setting "When to bill" -> "After validated order" or "After validated shipping sent"?

Reply:
@atm-maxime Can you change the test on  "facture = 1"  or if you prefer closing the PR ?

Reply:
@eldy I can update this PR. I think adding an option in the order module, just to display this workboard indicator will mislead the user in thinking that the option changes the workflow on orders.
What I suggest is to change the workboard like this :

TODAY :
![image](https://user-images.githubusercontent.com/2097773/229301563-d8aa1925-85a9-4abc-93bc-2a25aa6771aa.png)
(note that the link on the "open" orders is wrong, it uses the filter showing closed orders also, I'll fix this too)

AFTER :
![image](https://user-images.githubusercontent.com/2097773/229303883-c7d17280-f50a-4d4b-9f60-208b9c634447.png)
À livrer => status IN (1,2)
À facturer => status IN (1,2,3) AND facture = 0
Livrée / À facturer => status = 3 AND facture = 0

What do you think ?

Reply:
@eldy ?

Reply:
> @eldy I can update this PR. I think adding an option in the order module, just to display this workboard indicator will mislead the user in thinking that the option changes the workflow on orders. What I suggest is to change the workboard like this :
> 
> TODAY : ![image](https://user-images.githubusercontent.com/2097773/229301563-d8aa1925-85a9-4abc-93bc-2a25aa6771aa.png) (note that the link on the "open" orders is wrong, it uses the filter showing closed orders also, I'll fix this too)
> 
> AFTER : ![image](https://user-images.githubusercontent.com/2097773/229303883-c7d17280-f50a-4d4b-9f60-208b9c634447.png) À livrer => status IN (1,2) À facturer => status IN (1,2,3) AND facture = 0 Livrée / À facturer => status = 3 AND facture = 0
> 
> What do you think ?

Looks good to me.

Reply:
Done !